admiral/pkg/controller/common/config_test.go

```diff
@@ -308,6 +308,146 @@ func TestIsSlowStartEnabledForCluster(t *testing.T) {
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			p.EnableVSRoutingInCluster = tc.enableVSRoutingInCluster
+			p.VSRoutingInClusterEnabledClusters = tc.enabledVSRoutingInClusterForCluster
+			ResetSync()
+			InitializeConfig(p)
+
+			assert.Equal(t, tc.expected, DoVSRoutingInClusterForCluster(tc.cluster))
+		})
+	}
+
+}
+
+func TestDoVSRoutingInClusterForIdentity(t *testing.T) {
+	p := AdmiralParams{}
+
+	testCases := []struct {
+		name                                   string
+		identity                               string
+		enableVSRoutingInCluster               bool
+		enabledVSRoutingInClusterForIdentities []string
+		expected                               bool
+	}{
+		{
+			name: "Given enableVSRoutingInCluster is false, enabledVSRoutingInClusterForIdentities is empty" +
+				"When DoVSRoutingInClusterForIdentity is called" +
+				"Then it should return false",
+			identity:                               "testIdentity1",
+			enableVSRoutingInCluster:               false,
+			enabledVSRoutingInClusterForIdentities: []string{},
+			expected:                               false,
+		},
+		{
+			name: "Given enableVSRoutingInCluster is true, enabledVSRoutingInClusterForIdentities is empty" +
+				"When DoVSRoutingInClusterForIdentity is called" +
+				"Then it should return false",
+			identity:                               "testIdentity1",
+			enableVSRoutingInCluster:               true,
+			enabledVSRoutingInClusterForIdentities: []string{},
+			expected:                               false,
+		},
+		{
+			name: "Given enableVSRoutingInCluster is true, and given cluster doesn't exists in the list" +
+				"When DoVSRoutingInClusterForIdentity is called" +
+				"Then it should return false",
+			identity:                               "testIdentity2",
+			enableVSRoutingInCluster:               true,
+			enabledVSRoutingInClusterForIdentities: []string{"testIdentity1"},
+			expected:                               false,
+		},
+		{
+			name: "Given enableVSRoutingInCluster is true, and given cluster does exists in the list" +
+				"When DoVSRoutingInClusterForIdentity is called" +
+				"Then it should return true",
+			identity:                               "testIdentity1",
+			enableVSRoutingInCluster:               true,
+			enabledVSRoutingInClusterForIdentities: []string{"testIdentity1"},
+			expected:                               true,
+		},
+		{
+			name: "Given enableVSRoutingInCluster is true, and all VS routing is enabled in all clusters using '*'" +
+				"When DoVSRoutingInClusterForIdentity is called" +
+				"Then it should return true",
+			identity:                               "testIdentity1",
+			enableVSRoutingInCluster:               true,
+			enabledVSRoutingInClusterForIdentities: []string{"*"},
+			expected:                               true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			p.EnableVSRoutingInCluster = tc.enableVSRoutingInCluster
```

Comment:
https://github.com/istio-ecosystem/admiral/blob/master/admiral/pkg/clusters/handler_test.go#L26
https://github.com/istio-ecosystem/admiral/blob/master/admiral/pkg/clusters/clientdiscovery_handler_test.go#L249

should we use this approach for initializing admiral  instead of manually modifying struct?

Reply:
it is a nit pick so feel free to omit.

Reply:
@adilfulara , thats a good point but initAdmiral func does a lot more than what I need. Just wanted to keep it simple and set params that I need.